### PR TITLE
CM-24003 - Integrate Cimon

### PR DESCRIPTION
**Pull request by MarshalX**

### CODE REVIEW

**Comment by MarshalX** [comment]
@MaorDavidzon could you please setup required secrets? CIMON_CLIENT_ID and CIMON_SECRET (pls create personal token/service account too) .

File: .github/workflows/pre_release.yml [hunk 1]
```diff
@@ -14,41 +14,44 @@ jobs:
...
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: |
+          gh run cancel ${{ github.run_id }}
+          gh run watch ${{ github.run_id }}
+
+      - name: Build package
+        run: poetry build
+
+      - name: Publish a Python distribution to PyPI
+        uses: pypa/gh-action-pypi-publish@release/v1
```
[the remark this answers is not in the document]
Comment (MarshalX) [reply .github/workflows/pre_release.yml:57]: #cycode_ignore_insecure_pipeline_violation_here

File: .github/workflows/pre_release.yml [hunk 2]
```diff
@@ -14,41 +14,44 @@ jobs:
...
+        with:
+          fetch-depth: 0
+
+      - name: Set up Python 3.7
+        uses: actions/setup-python@v4
+        with:
+          python-version: '3.7'
+
+      - name: Setup Poetry
+        uses: snok/install-poetry@v1
```
[the remark this answers is not in the document]
Comment (MarshalX) [reply .github/workflows/pre_release.yml:34]: #cycode_ignore_insecure_pipeline_violation_here

File: .github/workflows/release.yml [hunk 3]
```diff
@@ -23,11 +29,8 @@ jobs:
         with:
           python-version: '3.7'
 
-      - name: Install Poetry
-        run: curl -sSL https://install.python-poetry.org | python - -y
-
-      - name: Update PATH
-        run: echo "$HOME/.local/bin" >> $GITHUB_PATH
+      - name: Setup Poetry
+        uses: snok/install-poetry@v1
```
[the remark this answers is not in the document]
Comment (MarshalX) [reply .github/workflows/release.yml:33]: #cycode_ignore_insecure_pipeline_violation_here

**Comment by MaorDavidzon** [comment]
> @MaorDavidzon could you please setup required secrets? `CIMON_CLIENT_ID` and `CIMON_SECRET` (pls create personal token/service account too).
> 
> And pls allow `cycodelabs/cimon-action@v0 is not allowed to be used in cycodehq-public/cycode-cli`

Done

**Review by MaorDavidzon — APPROVED** [review]

**Pull request merged**